src/public/services/AdminSubmissionsService.ts

```diff
@@ -65,3 +68,43 @@ export const getEncryptedResponse = ({
     .get(`${ADMIN_FORM_ENDPOINT}/${formId}/submissions/${submissionId}`)
     .then(({ data }) => data)
 }
+
+/**
+ * Triggers a download of a set of attachments as a zip file when given attachment metadata and a secret key
+ * @param {Map} attachmentDownloadUrls Map of question number to individual attachment metadata (object with url and filename properties)
+ * @param {String} secretKey An instance of EncryptionKey for decrypting the attachment
+ * @returns {Promise} A Promise containing the contents of the ZIP file as a blob
+ */
+export const downloadAndDecryptAttachmentsAsZip = async (
+  attachmentDownloadUrls: Map<number, { url: string; filename: string }>,
+  secretKey: string,
+): Promise<Blob> => {
+  const zip = new JSZip()
+  const downloadPromises = Array.from(attachmentDownloadUrls).map(
```

Comment:
why use `Array.from` instead of `Map.forEach`?

Reply:
oh actually cos i wanted to use `map` rather than `forEach` cos i wanted to make it clear that this is just a transformation @_@

can change in accordance with below

Reply:
ah okay sure, that makes sense

Reply:
sharing [this](https://jesseduffield.com/Array-Functions-And-Rule-Of-Least-Power/), which was the article that convinced me that `map` is (usually) better.

now i use `map` when it's solely a transformation and i want to signal that order doesn't matter